ev3dev/core.py

```diff
@@ -955,6 +958,94 @@ def wait_while(self, s, timeout=None):
         """
         return self.wait(lambda state: s not in state, timeout)
 
+    def set_power(self, power):
+        self.speed_sp = power
+
+    def set_position_rotations(self, power, rotations):
+        if power > 0:
+            self.position_sp = self.position + int(rotations * self.count_per_rot)
+        else:
+            self.position_sp = self.position - int(rotations * self.count_per_rot)
+
+    def set_position_degrees(self, power, degrees):
+        rotations = float(degrees / self.count_per_rot)
+        self.set_position_rotations(power, rotations)
+
+    def set_brake(self, brake):
+        if brake:
+            self.stop_action = self.STOP_ACTION_BRAKE
+        else:
+            self.stop_action = self.STOP_ACTION_COAST
+
+    def set_time(self, ms):
+        self.time_sp = ms
+
+    def on_for_rotations(self, power, rotations, brake):
+        """
+        Rotate the motor at 'power' for 'rotations'
+        """
+        self.set_power(power)
```

Comment:
These powers will be in tacho counts per second. Do we not want to take input on a [-100,100] scale as LEGO does?

Reply:
Should the scale be `[-1 * self.max_speed, self.max_speed]`?

Reply:
The stock Lego interface measures speed/power in percentage of `max_speed`, and the `max_speed` is never exposed.

Reply:
ah I see now...will clean this up

Reply:
I share the same sentiments as @laurensvalk that we should be using scientific units where possible in places where LEGO uses 0..100 or -100..100

https://github.com/ev3dev/ev3dev/issues/668#issuecomment-224670770

So, in this case, our speed parameter should be in degrees per second or RPM.

Reply:
Also, I wouldn't mined a default value for the brake parameter here as `True` since this is what gets used most of the time.

Reply:
wait if the EV3-G on_for_rotations block only takes power as a percentage of max_speed and we take power as the scientific unit then our attempt at a 1:1 API isn't going to match